src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -134,36 +146,43 @@ object DemuxFastqs {
     }
   }
 
-  /** Creates a demultiplexing iterator that performs demultiplexing in parallel.
+  /** Creates a demultiplexing iterator.
     *
     * @param sources the FASTQ sources, one per read.
     * @param demultiplexer the demultiplexer to use.  The demultiplexer's [[com.fulcrumgenomics.fastq.FastqDemultiplexer.demultiplex()]]
     *                      method expects the same number of reads as sources.
     */
-  def demultiplexingIterator(sources: Seq[FastqSource],
+  def demultiplexingIterator(sources: Seq[Iterator[FastqRecord]],
                              demultiplexer: FastqDemultiplexer,
-                             threads: Int,
-                             batchSize: Int = DemuxBatchRecordsSize): Iterator[DemuxResult] = {
-
+                             asyncOptions: DemuxFastqsAsyncOptions): Iterator[DemuxResult] = {
     require(demultiplexer.expectedNumberOfReads == sources.length,
       s"The demultiplexer expects ${demultiplexer.expectedNumberOfReads} reads but ${sources.length} FASTQ sources given.")
 
-    val zippedIterator = FastqSource.zipped(sources)
-    if (threads > 1) {
+    val demuxSource = asyncOptions.asyncReadingType match {
+      case DemuxFastqsAsyncReadingType.NoAsync         => FastqSource.zipped(sources)
+      case DemuxFastqsAsyncReadingType.SingleThread    => AsyncIterator(FastqSource.zipped(sources), bufferSize=Some(1024))
```

Comment:
Perhaps `1024` could either be a constant somewhere _or_ another value in the async options case class, or use the batch size from that class?